### PR TITLE
prepare release process to maven central

### LINKED ISSUE #11: setup a lightweight release process
Following this tutorial [1] the release process should be triggered by only adding a tag.

[1] https://dracoblue.net/dev/uploading-snapshots-and-releases-to-maven-central-with-travis/

- [ ] complete documentation
- [x] test with newly tagged version

### PR DESCRIPTION
**Pull request by lkwg82**
Following this tutorial [1] the release process should be triggered by only adding a tag. 

- [x] evaluate temporary gpg key creation and usage
- [x] adjust `.travis.yml` to release only from master
- [x] adjust version in `pom.xml` (this will not reflect the last version in maven central)
- [x] update readme?
- [x] activate autoclose in release profile, to automatic release after deployment

[1] https://dracoblue.net/dev/uploading-snapshots-and-releases-to-maven-central-with-travis/

should fix #11

### CODE REVIEW

**Comment by lkwg82** [comment]
see first successful beta release in https://travis-ci.org/idealo/logback-redis/builds/158578260

**Comment by lkwg82** [comment]
central sync activated: https://issues.sonatype.org/browse/OSSRH-24847

File: .travis/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ ! -z "$TRAVIS_TAG" ]
+then
+    echo "on a tag -> set pom.xml <version> to $TRAVIS_TAG"
+    mvn --settings .travis/settings.xml org.codehaus.mojo:versions-maven-plugin:2.1:set -DnewVersion=$TRAVIS_TAG -Prelease 1>/dev/null 2>/dev/null
+
```
Comment (cdanzl) [inline .travis/deploy.sh]: increase plugin version to 2.3, don't suppress stderr

File: .travis/gpg.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      Subkey-Length: 4096
+      Name-Real: Opensource Idealo
+      Name-Email: opensource-logback-redis@idealo.de
+      Expire-Date: 2y
+      Passphrase: ${GPG_PASSPHRASE}
+      %commit
+      %echo done
+EOF
+
+gpg --batch --gen-key gen-key-script 2>&1 | tail -n1 | cut -d\  -f3
```
Comment (cdanzl) [inline .travis/gpg.sh]: bitte säubern...

File: .travis/gpg.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+export GPG_KEYNAME=$(gpg -K | grep ^sec | cut -d/  -f2 | cut -d\  -f1 | head -n1)
+shred gen-key-script
+gpg --send-keys ${GPG_KEYNAME}
+while(true); do
```
Comment (cdanzl) [inline .travis/gpg.sh]: please comment

**Pull request merged**